x-pack/plugins/cloud_security_posture/public/pages/findings/utils.ts

```diff
@@ -49,6 +52,23 @@ export const getPaginationTableParams = (
   showPerPageOptions,
 });
 
+export const usePersistedQuery = <T>(getter: ({ filters, query }: FindingsBaseURLQuery) => T) => {
+  const {
+    data: {
+      query: { filterManager, queryString },
+    },
+  } = useKibana().services;
+
+  return useCallback(
+    () =>
+      getter({
+        filters: filterManager.getAppFilters(),
+        query: queryString.getQuery(),
+      }),
+    [getter, filterManager, queryString]
+  );
+};
+
```

Comment:
this is where we wrap each `getDefaultQuery` we get in a `useCallback` and inject it with current app filters